akka-javasdk/src/main/scala/akka/javasdk/impl/workflow/ReflectiveWorkflowRouter.scala

```diff
@@ -98,39 +93,27 @@ class ReflectiveWorkflowRouter[S, W <: Workflow[S]](
 
     val workflow = instanceFactory(workflowContext)
 
-    val commandEffect =
-      try {
-        // if runtime doesn't have a state to provide, we fall back to user's own defined empty state
-        val decodedState = decodeUserState(userState).getOrElse(workflow.emptyState())
-        workflow._internalSetup(decodedState, context, timerScheduler)
-
-        val methodInvoker = methodInvokerLookup(commandName)
-
-        if (serializer.isJson(command) || command.isEmpty) {
-          // - BytesPayload.empty - there is no real command, and we are calling a method with arity 0
-          // - BytesPayload with json - we deserialize it and call the method
-          val deserializedCommand =
-            CommandSerialization.deserializeComponentClientCommand(methodInvoker.method, command, serializer)
-          val result = deserializedCommand match {
-            case None        => methodInvoker.invoke(workflow)
-            case Some(input) => methodInvoker.invokeDirectly(workflow, input)
-          }
-          result.asInstanceOf[Workflow.Effect[_]]
-        } else {
-          throw new IllegalStateException(
-            s"Could not find a matching command handler for method [$commandName], content type [${command.contentType}] " +
-            s"on [${workflow.getClass.getName}]")
-        }
-
-      } catch {
-        case CommandHandlerNotFound(name) =>
```

Comment:
was this dead-code?

Reply:
yeap, I simplified this code before, and nothing throws this exception anymore